pkg/operator/encryption/controllers.go

```diff
@@ -47,6 +51,17 @@ func NewControllers(
 		return nil, err
 	}
 
+	// for testing resourceSyncer might be nil
+	if resourceSyncer != nil {
+		if err := resourceSyncer.SyncSecretConditionally(
```

Comment:
the previous location was also okay.

Reply:
we don't have all the information needed by the precondition in the deployer. We would have to pass new parameters just for that so I think it is better to just move the logic to the encryption controller.